docs/guides/debugging.md

```diff
@@ -0,0 +1,226 @@
+```
+
+Let's fire up our integration test with `earth -P -i +test`:
+
+```
+buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
++base | --> FROM python:3
+context | --> local context .
++base | resolve docker.io/library/python:3@sha256:e9b7e3b4e9569808066c5901b8a9ad315a9f14ae8d3949ece22ae339fff2cad0 100%
+context | transferring .: 100%
++base | *cached* --> WORKDIR /code
++server | *cached* --> COPY server.py .
++test | --> FROM docker:19.03.12-dind
++test | resolve docker.io/library/docker:19.03.12-dind@sha256:674f1f40ff7c8ac14f5d8b6b28d8fb1f182647ff75304d018003f1e21a0d8771 100%
++test | *cached* --> RUN apk add curl
++test | --> WITH DOCKER RUN docker run --rm -d --network=host server:latest python3 server.py && sleep 5 && curl -s localhost:8000 | grep hello
++test | Loading images...
++test | Loaded image: server:latest
++test | ...done
++test | 1dc054c647cb75bde4897a2828edb095739cb9f864ed203ed2ddb54e62554aad
++test | Command /bin/sh -c docker run --rm -d --network=host server:latest python3 server.py && sleep 5 && curl -s localhost:8000 | grep hello failed with exit code 1
++test | Entering interactive debugger (**Warning: only a single debugger per host is supported**)
+```
+
+
+
+There was a failure checking that the server output contained the string `hello`; let's see what is going on:
+
+
+```
+/ # docker ps -a
+CONTAINER ID        IMAGE               COMMAND               CREATED             STATUS              PORTS               NAMES
+b8a31c54dd17        server:latest       "python3 server.py"   5 seconds ago       Up 4 seconds                            frosty_rhodes
+```
+
+The good news is our server container is running; let's see what happens when we try to connect to it:
+
+```
+/ # curl -s localhost:8000
+Hello, world!/ 
+```
+
+Ah ha! The problem is our test is expecting a lowercase `h`, so we can fix our grep to look for an uppercase `H`:
+
+```Dockerfile
+# Earthfile
+
+server:
+  COPY server.py .
+  SAVE IMAGE
+
+test:
+  FROM docker:19.03.12-dind
+  RUN apk add curl
+  WITH DOCKER
+    DOCKER LOAD +server server:latest
+    RUN docker run --rm -d --network=host server:latest python3 server.py && sleep 5 && curl -s localhost:8000 | grep Hello
+  END
+```
+
+Then when we re-run our test we get:
+
+```
++test | --> WITH DOCKER RUN docker run --rm -d --network=host server:latest python3 server.py && sleep 5 && curl -s localhost:8000 | grep Hello
++test | Loading images...
++test | Loaded image: server:latest
++test | ...done
++test | cb5299ae03cd17cfb2b528f01268ccf59761feec036cb313a3e969930d6f0815
++test | Hello, world!
++test | Target +test built successfully
+=========================== SUCCESS ===========================
+```
+
+With the use of the interactive debugger; we were able to examine the state of the embeded containerized 
+
+## Final tips
+
+If you ever want to jump into an interactive debugging session at any point in your Earthfile, you can simply add a command that will fail such as:
```

Comment:
We should also remind them to use `-i`, besides inserting `RUN false` - just in case they scan this guide and miss that they need to add that flag.

Reply:
good point, that's been added now: `and run earth with the `--interactive` (or `-i`) flag.`